pkg/kube/kube_handler_test.go

```diff
@@ -331,6 +331,58 @@ func TestAppRepositoryCreate(t *testing.T) {
 	}
 }
 
+func TestAppRepositoryList(t *testing.T) {
+	testCases := []struct {
+		name             string
+		requestNamespace string
+		existingRepos    map[string][]repoStub
+	}{
+		{
+			name:             "it gets repos from the global namespace",
+			requestNamespace: kubeappsNamespace,
+			existingRepos: map[string][]repoStub{
+				"kubeapps": {repoStub{name: "test-repo"}},
+			},
+		},
+		{
+			name:             "it gets repos from a namespace",
+			requestNamespace: "foo",
+			existingRepos: map[string][]repoStub{
+				"foo": {repoStub{name: "test-repo"}},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			cs := fakeCombinedClientset{
+				fakeapprepoclientset.NewSimpleClientset(makeAppRepoObjects(tc.existingRepos)...),
+				fakecoreclientset.NewSimpleClientset(),
+				&fakeRest.RESTClient{},
+			}
+			handler := userHandler{
+				kubeappsNamespace: kubeappsNamespace,
+				svcClientset:      cs,
+			}
+			if tc.requestNamespace != kubeappsNamespace {
+				handler = userHandler{
+					kubeappsNamespace: kubeappsNamespace,
+					clientset:         cs,
+				}
+			}
```

Comment:
This seems backwards? The test should not be setting either the `svcClientset` or the `clientset`, but rather, it should set both (perhaps with different fake data?) and ensuring that the code uses the correct handler.

Hmm, actually, I think I see your intention: the code would error here if it chose the incorrect clientset, so you're setting either one or the other to ensure an error is raised if the incorrect code path went? It's a bit fragile, but ok - not backwards as I initially thought. Perhaps a comment so the intention is clear, if that is the intention?

Reply:
Yes, you got the intention, the goal here is to fail if the test is trying to use the wrong clientset, will add a comment about that.